include/rocksdb/env.h

```diff
@@ -530,12 +533,19 @@ class Env {
 
   virtual void SanitizeEnvOptions(EnvOptions* /*env_opts*/) const {}
 
+  // Get the FileSystem implementation this Env was constructed with. It
+  // could be a fully implemented one, or a wrapper class around the Env
+  std::shared_ptr<FileSystem> GetFileSystem();
+
   // If you're adding methods here, remember to add them to EnvWrapper too.
 
  protected:
   // The pointer to an internal structure that will update the
   // status of each thread.
   ThreadStatusUpdater* thread_status_updater_;
+
+  // Pointer to the underlying FileSystem implementation
+  std::shared_ptr<FileSystem> file_system_;
```

Comment:
What is the difference between `Env::file_system_` and `CompositeEnv::fs_env_ptr`?

Reply:
```CompositeEnv``` can use ```Env::file_system_```. I believe there's no need for ```fs_env_ptr_```.